<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>630</width>
    <height>580</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mission Control</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetNoConstraint</enum>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Altitude (meters)</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLCDNumber" name="altitudeLCD">
          <property name="accessibleName">
           <string>Altitude</string>
          </property>
          <property name="accessibleDescription">
           <string>Altitude</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="digitCount">
           <number>7</number>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::Flat</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Velocity (m/s)</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLCDNumber" name="lcdNumber">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="smallDecimalPoint">
           <bool>false</bool>
          </property>
          <property name="digitCount">
           <number>7</number>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::Flat</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <widget class="QLabel" name="label_6">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Temperature (ÂºC)</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLCDNumber" name="lcdNumber_2">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="digitCount">
           <number>7</number>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::Flat</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Pitch</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
            <property name="indent">
             <number>-1</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLCDNumber" name="pitchLCD">
            <property name="whatsThis">
             <string>Pitch</string>
            </property>
            <property name="accessibleName">
             <string>Pitch</string>
            </property>
            <property name="accessibleDescription">
             <string>Pitch</string>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="segmentStyle">
             <enum>QLCDNumber::Flat</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Roll</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLCDNumber" name="rollLCD">
            <property name="whatsThis">
             <string>Roll</string>
            </property>
            <property name="accessibleName">
             <string>Roll</string>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="segmentStyle">
             <enum>QLCDNumber::Flat</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Yaw</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLCDNumber" name="yawLCD">
            <property name="whatsThis">
             <string>Yaw</string>
            </property>
            <property name="accessibleName">
             <string>Yaw</string>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="segmentStyle">
             <enum>QLCDNumber::Flat</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>ECHO</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_9">
      <item>
       <widget class="QCustomPlot" name="altitudePlot" native="true"/>
      </item>
      <item>
       <widget class="QCustomPlot" name="velocityPlot" native="true"/>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLCDNumber" name="latitudeLCD">
          <property name="whatsThis">
           <string>Latitude</string>
          </property>
          <property name="accessibleName">
           <string>Latitude</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::Flat</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLCDNumber" name="longitudeLCD">
          <property name="whatsThis">
           <string>Longitude</string>
          </property>
          <property name="accessibleName">
           <string>Longitude</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::Flat</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QWebEngineView" name="webEngineView" native="true">
        <property name="url" stdset="0">
         <url>
          <string>about:blank</string>
         </url>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QWebEngineView</class>
   <extends>QWidget</extends>
   <header location="global">QtWebEngineWidgets/QWebEngineView</header>
  </customwidget>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateAltitude(double)</signal>
   <receiver>altitudeLCD</receiver>
   <slot>display(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>61</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>61</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateVelocity(double)</signal>
   <receiver>lcdNumber</receiver>
   <slot>display(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>8</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>226</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateTemperature(double)</signal>
   <receiver>lcdNumber_2</receiver>
   <slot>display(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>7</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updatePitch(double)</signal>
   <receiver>pitchLCD</receiver>
   <slot>display(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1</x>
     <y>516</y>
    </hint>
    <hint type="destinationlabel">
     <x>39</x>
     <y>516</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateRoll(double)</signal>
   <receiver>rollLCD</receiver>
   <slot>display(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>8</x>
     <y>437</y>
    </hint>
    <hint type="destinationlabel">
     <x>131</x>
     <y>523</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateYaw(double)</signal>
   <receiver>yawLCD</receiver>
   <slot>display(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>132</x>
     <y>573</y>
    </hint>
    <hint type="destinationlabel">
     <x>235</x>
     <y>523</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateLatitude(double)</signal>
   <receiver>latitudeLCD</receiver>
   <slot>display(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>464</x>
     <y>1</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateLongitude(double)</signal>
   <receiver>longitudeLCD</receiver>
   <slot>display(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>622</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>596</x>
     <y>57</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>echo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>551</y>
    </hint>
    <hint type="destinationlabel">
     <x>249</x>
     <y>546</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>updateAltitude(double)</signal>
  <signal>updateVelocity(double)</signal>
  <signal>updateTemperature(double)</signal>
  <signal>updatePitch(double)</signal>
  <signal>updateRoll(double)</signal>
  <signal>updateYaw(double)</signal>
  <signal>updateLatitude(double)</signal>
  <signal>updateLongitude(double)</signal>
  <signal>newMessage(QString)</signal>
  <slot>echo()</slot>
 </slots>
</ui>
